modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/workflow/WorkflowApprovalExecutarTest.java

```diff
@@ -0,0 +1,913 @@
+        // Gets the new configuration of the workflow-extentions.xml
+        newWFExtentionsXML = readFile(getAMResourceLocation() + File.separator + "configFiles" + File.separator
+                + "approveWorkflow" + File.separator + "workflow-extentions.xml");
+        // Updates the content of the workflow-extentions.xml of the registry file, to have the new configurations.
+        resourceAdminServiceClient.updateTextContent(DEFAULT_WF_EXTENTIONS_XML_REG_CONFIG_LOCATION, newWFExtentionsXML);
+
+        apiPublisher = new APIPublisherRestClient(publisherUrls.getWebAppURLHttp());
+        apiStore = new APIStoreRestClient(storeUrls.getWebAppURLHttp());
+    }
+
+    @Test(groups = { "wso2.am" }, description = "Api workflow process check")
+    public void apiWorkflowProcessCheck() throws Exception {
+
+        //add API
+        String apiName = "WorkflowCheckAPI";
+        String apiVersion = "1.0.0";
+        String apiContext = "workflowCheck";
+        String url = getGatewayURLHttp() + "jaxrs_basic/services/customers/customerservice";
+
+        APIRequest apiRequest;
+        apiRequest = new APIRequest(apiName, apiContext, new URL(url));
+
+        apiRequest.setVersion(apiVersion);
+        apiRequest.setTiersCollection(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setTier(APIMIntegrationConstants.API_TIER.UNLIMITED);
+        apiRequest.setProvider(USER_SMITH);
+
+        HttpResponse apiResponse = restAPIPublisher.addAPI(apiRequest);
+        apiId= apiResponse.getData();
+        //request to publish the API
+        HttpResponse lifeCycleChangeResponse = restAPIPublisher
+                .changeAPILifeCycleStatus(apiId, APILifeCycleAction.PUBLISH.getAction(), null);
+
+
+        assertEquals(apiResponse.getResponseCode(), 201,
+                "API creation test failed in Approval Workflow Executor");
+        assertEquals(lifeCycleChangeResponse.getResponseCode(), 200,
+                "Status code mismatch when request is make for change API state");
+
+        // check the current state
+        HttpResponse api = restAPIPublisher.getAPI(apiId);
+        JSONObject apiObj = new JSONObject(api.getData());
+        String apiStatus = (String) apiObj.get("workflowStatus");
+
+        // lifecycle state should not change
+        assertEquals(apiStatus, APILifeCycleState.CREATED.toString(),
+                "Lifecycle state should remain without changing till approval. ");
+
+        //Get workflow pending requests by admin
+        String workflowType = "AM_API_STATE";
+        HttpResponse response1 = restAPIAdmin.getWorkflows(workflowType);
+        assertEquals(response1.getResponseCode(), 200,
+                "Get Workflow Pending requests failed for User Admin");
+
+        //Get workflow pending requests by unauthorized user
+        restAPIStoreFirstUser = new RestAPIAdminImpl(USER_SMITH, "john123", "carbon.super" , "https://localhost:9943/");
```

Comment:
don't use hardcoded urls. Also I think you can define this in the setenv and use it throughout the test